# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/x11/xorg/startwm.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

More flexible root application startup. Kind of ROCK Linux specific.

--- ./programs/xdm/config/Xsetup_0.orig	Wed Apr 27 09:20:03 1994
+++ ./programs/xdm/config/Xsetup_0	Thu Dec 28 15:23:15 2000
@@ -1,3 +1,4 @@
 #!/bin/sh
 # $XConsortium: Xsetup_0,v 1.3 93/09/28 14:30:31 gildea Exp $
 xconsole -geometry 480x130-0-0 -daemon -notify -verbose -fn fixed -exitOnFail
+xsetroot -solid black
--- ./programs/xdm/config/Xsession.cpp.orig	2006-03-18 22:31:35.000000000 +0000
+++ ./programs/xdm/config/Xsession.cpp	2006-03-18 22:33:23.000000000 +0000
@@ -56,5 +56,14 @@
                 exec `eval $XDESKTOP`
         }
 #endif
+
+XCOMM we need at least the PATH /opt/.../bin expansion ...
+	[ -f /etc/profile ] && . /etc/profile
+
+	WINDOWMANAGER="${WINDOWMANAGER:-startkde}"
+	[ "$WINDOWMANAGER" =  "gnome" ] && WINDOWMANAGER="gnome-session"
+	[ "$WINDOWMANAGER" =  "kde"   ] && WINDOWMANAGER="startkde"
+	[ "$WINDOWMANAGER" != "twm" -a -n "$(type -p ${WINDOWMANAGER%% *})" ] && exec $WINDOWMANAGER
+	xsetroot -solid darkcyan ; exec twm ; exec xterm
 	exec BINDIR/xsm
 fi
--- ./programs/xinit/xinitrc.cpp.orig	2006-03-18 22:36:41.000000000 +0000
+++ ./programs/xinit/xinitrc.cpp	2006-03-18 22:38:20.000000000 +0000
@@ -24,7 +24,7 @@
     XMODMAP $usermodmap
 fi
 
-XCOMM start some nice programs
+XCOMM start the window manager
 
 #if defined(__SCO__) || defined(__UNIXWARE__)
 if [ -r /etc/default/xdesktops ]; then
@@ -68,8 +68,8 @@
 
 XCOMM This is the fallback case if nothing else is executed above
 #endif /* !defined(__SCO__)  && !defined(__UNIXWARE__) */
-TWM &
-XCLOCK -geometry 50x50-1+1 &
-XTERM -geometry 80x50+494+51 &
-XTERM -geometry 80x20+494-0 &
-exec XTERM -geometry 80x66+0+0 -name login
+WINDOWMANAGER="${WINDOWMANAGER:-startkde}"
+[ "$WINDOWMANAGER" =  "gnome" ] && WINDOWMANAGER="gnome-session"
+[ "$WINDOWMANAGER" =  "kde"   ] && WINDOWMANAGER="startkde"
+[ "$WINDOWMANAGER" != "twm" -a -n "$(type -p ${WINDOWMANAGER%% *})" ] && exec $WINDOWMANAGER
+xsetroot -solid darkcyan ; exec twm ; exec xterm
